pkg/utils/environment/kernel_symbols.go

```diff
@@ -22,110 +22,150 @@ type KernelSymbol struct {
 	Address uint64
 	Owner   string
 }
-
-func symNotFoundErr(v interface{}) error {
-	return fmt.Errorf("symbol not found: %v", v)
-}
-
-//
-// Interface implementation
-//
-
-type name struct {
-	name string
-}
-
-type addr struct {
-	addr uint64
-}
-
 type nameAndOwner struct {
 	name  string
 	owner string
 }
-
 type addrAndOwner struct {
 	addr  uint64
 	owner string
 }
 
+// KernelSymbolTable manages kernel symbols with multiple maps for fast lookup.
 type KernelSymbolTable struct {
```

Comment:
I think we may want to add the option to batch require and query symbols. This would be useful in cases like `hooked_symbols` where we may receive multiple new symbols to query at once. The current API would have to make much more rescans than is necessary, since we can batch request all the new symbols at once, and only then trigger the scan.

Reply:
This can be a future optimization in a different PR.
For now, it is critical to resolve the memory footprint issue

Reply:
I was likely mistaken about having a relevant case for this right now in the codebase. Though it could be a possibility in the future, so it would still be good to add this API (in another PR), so that we don't accidentally introduce this regression in the future.

Anyway, resolving this one.